### DISCUSSION TITLE
Enum variables not working: Only scalar input variables are allowed.

### DISCUSSION THREAD

**Discussion by jnnkB** [geldata/gel#2413] [Q&A] [answered]
Hey, I really enjoy using edgedb so far. But now I'm facing a minor issue. I'm trying to use enums as variables but I get the following error message:

> QueryError: Only scalar input variables are allowed. Variable 'category' has non-scalar value.

As far as I understood the documentation enums are scalars. At least they're listed there. So is there any other way I could do this? Or is this a bug/yet to be implemented?

Here is an example query:

```graphql
mutation updateItem($category: Category) {
  update_Item(data: {category: {set: $category}}) {
    category
  }
}
```

And an example for the variables:
```json 
{
  "category": "ENUM"
}
```

**Comment by 1st1** [comment, marked as answer]
@vpetrovykh I think this is a bug.

**Reply by vpetrovykh** [reply]
This indeed appears to be a bug, I opened an issue for it: https://github.com/edgedb/edgedb/issues/2415
